rusk/src/lib/state.rs

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::error::Error;
+use crate::Result;
+use dusk_bls12_381::BlsScalar;
+use dusk_pki::{Ownable, ViewKey};
+use dusk_poseidon::tree::PoseidonBranch;
+use microkelvin::{Backend, BackendCtor};
+use phoenix_core::Note;
+use rusk_abi::{self, POSEIDON_TREE_DEPTH};
+use rusk_vm::{NetworkState, NetworkStateId};
+use transfer_contract::TransferContract;
+
+pub struct RuskState(pub(crate) NetworkState);
+
+impl RuskState {
```

Comment:
We might want to add a `check_nullifier_exists` as a state method

Reply:
I filed issue #459 that should be used for that.